portfolio/src/main/java/com/google/sps/servlets/UserDataServlet.java

```diff
@@ -24,12 +24,45 @@
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.FetchOptions;
 import com.google.gson.Gson;
 import com.google.sps.data.UserAuth;
+import java.util.*;
 
 /** Servlet that communicates user information */
 @WebServlet("/user-data")
 public class UserDataServlet extends HttpServlet {
+    
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("text/plain; charset=UTF-8");
+
+        UserService userService = UserServiceFactory.getUserService();
+        if (!userService.isUserLoggedIn()) {
+            response.getWriter().println("");
+            return;
+        }
+
+        String userId = userService.getCurrentUser().getUserId();
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Filter propertyFilter = new FilterPredicate("id", FilterOperator.EQUAL, userId);
+        Query query = new Query("UserData").setFilter(propertyFilter);
+        PreparedQuery results = datastore.prepare(query);
+        List<Entity> listResults = results.asList(FetchOptions.Builder.withDefaults());
+
+        if (listResults.size() != 1) {
```

Comment:
Is it possible user id is not unique and we get 2 results?

Reply:
I'm pretty sure user IDs are unique unless something is manually changed on DataStore. I chose to use `listResults.size() != 1`  as the conditional mainly to ensure that the query is not empty, as well as to catch the unexpected case that the user ID is not unique.